### DISCUSSION TITLE
Shortcut for "foreach... yield return" pattern

### DISCUSSION THREAD

**Discussion by ekolis** [dotnet/csharplang#4169] [Language Ideas] [answered]
If you have a method which returns `IEnumerable<T>`, you currently have two options for how to return values: either as a "whole" enumerable such as an array or list:

```
public IEnumerable<int> CountToThree()
{
    return new int[]{1, 2, 3);
}
```

Or using `yield return` on individual values:

```
public IEnumerable<int> CountToThree()
{
    yield return 1;
    yield return 2;
    yield return 3;
}
```

However this becomes awkward when you want to combine the two:

```
public IEnumerable<int> CountToThree()
{
    yield return 1;
    yield return 2;
    yield return 3;
}

public int CountingStartsAtZero()
{
   return 0;
}

public IEnumerable<int> CountToThreeCorrectly()
{
     yield return CountingStartsAtZero();
     foreach (var i in CountToThree())
        yield return i;
}
```

I know this is a bit of a contrived example but I'm sure you can see practical scenarios where this would happen!

So there really ought to be some sort of shortcut for this "foreach... yield return" pattern. Maybe something like this:

```
public IEnumerable<int> CountToThreeCorrectly()
{
    yield return CountingStartsAtZero();
    yield return all CountToThree();
}
```

**Comment by CyrusNajmabadi** [comment, marked as answer]
Dup of https://github.com/dotnet/csharplang/discussions/378
